src/loophole.coffee

```diff
@@ -2,19 +2,35 @@ vm = require 'vm'
 
 exports.allowUnsafeEval = (fn) ->
   previousEval = global.eval
+  callback = -> global.eval = previousEval
+  error = false
   try
     global.eval = (source) -> vm.runInThisContext(source)
-    fn()
+    fn(callback)
```

Comment:
Basically I put both `sync` and `async` in the same `try..catch` block. but an extra var `error = false` is unfortunate. not sure if we could simplify this.